lib/filesystem.js

```diff
@@ -40,6 +40,63 @@ const MAXIMUM_RETRIES = 10;
 const RETRY_BASE_TIMEOUT = 100;
 
 /**
+ * @summary Handle read/write errors
+ * @function
+ * @private
+ *
+ * @param {Error} error - error
+ * @param {Object} options - options
+ * @param {Number} options.currentRetries - current number of retries
+ * @param {Function} options.retry - retry function (currentRetries)
+ * @param {Function} callback - callback (error)
+ *
+ * @example
+ * const foo = (callback, retries = 1) => {
+ *   bar((error) => {
+ *     if (error) {
+ *       handleError(error, {
+ *         currentRetries: retries,
+ *         retry: (currentRetries) => {
+ *           foo(callback, currentRetries + 1);
+ *         }
+ *       }, callback);
+ *     }
+ *
+ *     callback(null, 'Success!');
+ *   });
+ * };
+ */
+const handleError = (error, options, callback) => {
+  const platform = os.platform();
+
+  if (platform === 'darwin' && error.code === 'ENXIO') {
+    return callback(errors.unpluggedError());
+  }
+
+  if (platform === 'win32' && (error.code === 'ENOENT' || error.code === 'UNKNOWN')) {
+    return callback(errors.unpluggedError());
+  }
+
+  // In some faulty SDCards or SDCard readers you might randomly
+  // get EIO errors, but they usually go away after some retries.
+  if (error.code === 'EIO') {
+    if (options.currentRetries > MAXIMUM_RETRIES) {
+      if (platform === 'linux') {
+        return callback(errors.unpluggedError());
+      }
+
+      return callback(error);
```

Comment:
Yeah, was just thinking about that for ultimate correctness of the docs, as the callbacks might actually returns something else, but shouldn't really matter